// @gilbarbara/components/lib/Pagination/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Pagination = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("@emotion/react");
const styled_1 = require("@emotion/styled");
const is_lite_1 = require("is-lite");
const Button_1 = require("./Button");
const Icon_1 = require("../Icon");
const system_1 = require("../modules/system");
const StyledPagination = (0, styled_1.default)('div', (0, system_1.getStyledOptions)())(props => {
    const { align } = props;
    return (0, react_1.css) `
    align-items: center;
    display: flex;
    justify-content: ${align};
    ${(0, system_1.borderStyles)(props)};
    ${(0, system_1.marginStyles)(props)};
    ${(0, system_1.paddingStyles)(props)};
  `;
});
function Pagination(props) {
    const { currentPage, disableEdgeNavigation, edgeNavigationLimit = 3, onClick, totalPages, ...rest } = props;
    const items = [];
    if (totalPages <= 1) {
        return null;
    }
    if (!disableEdgeNavigation && totalPages > edgeNavigationLimit) {
        items.push({
            key: 'first',
            disabled: currentPage === 1,
            page: 1,
            content: (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: "chevron-double-left", size: 24 }),
        });
    }
    items.push({
        key: 'previous',
        disabled: currentPage === 1,
        page: currentPage - 1,
        content: (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: "chevron-left", size: 24 }),
    });
    if (disableEdgeNavigation && totalPages > 6) {
        items.push({ key: 'back', page: 1 });
        if (currentPage > 3) {
            items.push({ key: 'between-1', content: '...' });
        }
        if (currentPage === totalPages) {
            items.push({ page: currentPage - 2 });
        }
        if (currentPage > 2) {
            items.push({ page: currentPage - 1 });
        }
        if (currentPage !== 1 && currentPage !== totalPages) {
            items.push({ page: currentPage });
        }
        if (currentPage < totalPages - 1) {
            items.push({ page: currentPage + 1 });
        }
        if (currentPage === 1) {
            items.push({ page: currentPage + 2 });
        }
        if (currentPage < totalPages - 2) {
            items.push({ key: 'between-2', content: '...' });
        }
        items.push({ page: totalPages });
    }
    else {
        const pages = Array.from({ length: totalPages }, (_, index) => index + 1).filter(p => {
            const limit = currentPage + 2 <= totalPages ? currentPage - 1 : totalPages - 2;
            return (p >= limit &&
                p < currentPage + (currentPage === 1 || currentPage === totalPages - 1 ? 3 : 2));
        });
        pages.forEach(d => {
            items.push({ page: d });
        });
    }
    items.push({
        key: 'next',
        disabled: currentPage === totalPages,
        page: currentPage + 1,
        content: (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: "chevron-right", size: 24 }),
    });
    if (!disableEdgeNavigation && totalPages > edgeNavigationLimit) {
        items.push({
            key: 'last',
            disabled: currentPage === totalPages,
            page: totalPages,
            content: (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: "chevron-double-right", size: 24 }),
        });
    }
    return ((0, jsx_runtime_1.jsx)(StyledPagination, { "data-component-name": "Pagination", ...rest, children: items.map((d, index) => !is_lite_1.default.undefined(d.page) ? ((0, jsx_runtime_1.jsx)(Button_1.default, { currentPage: currentPage, disabled: d.disabled || false, onClick: onClick, page: d.page, children: d.content || d.page }, d.key || d.page || index)) : ((0, jsx_runtime_1.jsx)("span", { children: "..." }, d.key))) }));
}
exports.Pagination = Pagination;
Pagination.defaultProps = {
    align: 'end',
    disableEdgeNavigation: false,
    edgeNavigationLimit: 3,
};
