// @gilbarbara/components/lib/ButtonGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ButtonGroup = exports.StyledButtonGroup = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const react_2 = require("@emotion/react");
const styled_1 = require("@emotion/styled");
const system_1 = require("./modules/system");
exports.StyledButtonGroup = (0, styled_1.default)('div', (0, system_1.getStyledOptions)())(props => (0, react_2.css) `
    ${(0, system_1.baseStyles)(props)};
    display: inline-flex;

    > button {
      + button {
        margin-left: -1px;
      }

      &:first-of-type {
        border-bottom-right-radius: 0;
        border-top-right-radius: 0;
      }

      &:last-of-type {
        border-bottom-left-radius: 0;
        border-top-left-radius: 0;
      }

      &:not(:first-of-type):not(:last-of-type) {
        border-radius: 0;
      }
    }
  `);
function ButtonGroup(props) {
    const { children, shade, size, variant, ...rest } = props;
    const buttonProps = {
        shade,
        size,
        variant,
    };
    return ((0, jsx_runtime_1.jsx)(exports.StyledButtonGroup, { "data-component-name": "ButtonGroup", ...rest, children: react_1.Children.map(children, child => (0, react_1.cloneElement)(child, { ...buttonProps })) }));
}
exports.ButtonGroup = ButtonGroup;
ButtonGroup.defaultProps = {
    shade: 'mid',
    size: 'md',
    variant: 'primary',
};
