// @gilbarbara/components/lib/ButtonBase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ButtonBase = exports.StyledButtonBase = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const react_2 = require("@emotion/react");
const styled_1 = require("@emotion/styled");
const Icon_1 = require("./Icon");
const system_1 = require("./modules/system");
exports.StyledButtonBase = (0, styled_1.default)('button', (0, system_1.getStyledOptions)())(props => {
    const { busy } = props;
    return (0, react_2.css) `
    ${system_1.appearanceStyles};
    ${(0, system_1.baseStyles)(props)};
    align-items: center;
    background-color: transparent;
    border: 0;
    color: inherit;
    cursor: pointer;
    display: inline-flex;
    font-family: inherit;
    line-height: 1;
    padding: 0;
    ${(0, system_1.colorStyles)(props)};
    ${(0, system_1.marginStyles)(props)};
    ${(0, system_1.paddingStyles)(props)};
    ${(0, system_1.textStyles)(props)};

    :disabled {
      cursor: not-allowed;
      opacity: 0.6;
    }

    ${!!busy &&
        (0, react_2.css) `
      pointer-events: none;
    `};
  `;
});
exports.ButtonBase = (0, react_1.forwardRef)((props, ref) => {
    const { busy, children } = props;
    const { fontSize = '16px' } = (0, system_1.textStyles)(props);
    return ((0, jsx_runtime_1.jsxs)(exports.StyledButtonBase, { ref: ref, "data-component-name": "ButtonBase", ...props, children: [children, busy && (0, jsx_runtime_1.jsx)(Icon_1.Icon, { ml: "xxs", name: "spinner", size: parseInt(`${fontSize}`, 10) + 2, spin: true })] }));
});
exports.ButtonBase.defaultProps = {
    bold: false,
    busy: false,
    disabled: false,
    size: 'regular',
    type: 'button',
};
