// @gilbarbara/components/lib/ClickOutside.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClickOutsideMemo = exports.ClickOutside = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const DISPLAY = {
    BLOCK: 'block',
    FLEX: 'flex',
    INLINE: 'inline',
    INLINE_BLOCK: 'inline-block',
    CONTENTS: 'contents',
};
function ClickOutside(props) {
    const { active, children, display = DISPLAY.BLOCK, onClick, ...rest } = props;
    const containerRef = (0, react_1.useRef)(null);
    const isTouch = (0, react_1.useRef)(false);
    const handleClick = (0, react_1.useRef)((event) => {
        const container = containerRef.current;
        if (event.type === 'touchend') {
            isTouch.current = true;
        }
        if (event.type === 'click' && isTouch.current) {
            return;
        }
        if (container && !container.contains(event.target)) {
            onClick();
        }
    });
    (0, react_1.useEffect)(() => {
        const { current } = handleClick;
        if (active) {
            document.addEventListener('touchend', current, true);
            document.addEventListener('click', current, true);
        }
        return () => {
            document.removeEventListener('touchend', current, true);
            document.removeEventListener('click', current, true);
        };
    }, [active]);
    return ((0, jsx_runtime_1.jsx)("div", { ref: containerRef, "data-component-name": "ClickOutside", style: display !== DISPLAY.BLOCK && Object.values(DISPLAY).includes(display)
            ? { display }
            : undefined, ...rest, children: children }));
}
exports.ClickOutside = ClickOutside;
exports.ClickOutsideMemo = (0, react_1.memo)(ClickOutside);
