// @gilbarbara/components/lib/NonIdealState.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NonIdealState = exports.StyledNonIdealState = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("@emotion/react");
const styled_1 = require("@emotion/styled");
const helpers_1 = require("@gilbarbara/helpers");
const Box_1 = require("./Box");
const Headings_1 = require("./Headings");
const Icon_1 = require("./Icon");
const system_1 = require("./modules/system");
const Paragraph_1 = require("./Paragraph");
exports.StyledNonIdealState = (0, styled_1.default)('div', (0, system_1.getStyledOptions)('type'))(props => {
    const { direction } = props;
    return (0, react_1.css) `
    ${direction === 'horizontal' ? 'align-items: center;' : ''};
    display: ${direction === 'horizontal' ? 'flex' : 'block'};
    margin: 0 auto;
    text-align: ${direction === 'horizontal' ? 'left' : 'center'};
    width: 100%;
    ${(0, system_1.baseStyles)(props)};
    ${(0, system_1.backgroundStyles)(props, false)};
    ${(0, system_1.borderStyles)(props)};
    ${(0, system_1.flexItemStyles)(props)};
    ${(0, system_1.layoutStyles)(props)};
    ${(0, system_1.marginStyles)(props)};
    ${(0, system_1.paddingStyles)(props)};
    ${(0, system_1.radiusStyles)(props)};
    ${(0, system_1.shadowStyles)(props)};
  `;
});
function NonIdealState(props) {
    const { children, description, direction, hideIcon, icon, size = 'md', title, type } = props;
    const iconSize = {
        sm: 48,
        md: 64,
        lg: 96,
    };
    const isVertical = direction === 'vertical';
    const template = {};
    const output = {};
    switch (type) {
        case 'error': {
            template.icon = (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: "danger", size: iconSize[size] });
            template.title = 'Something went wrong';
            template.description = 'An unexpected error has occurred. Try reloading the page.';
            break;
        }
        case 'no-results': {
            template.icon = (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: "search", size: iconSize[size] });
            template.title = 'No search results';
            template.description = "Your search didn't match anything.";
            break;
        }
        case 'not-found': {
            template.icon = (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: "smile-none", size: iconSize[size] });
            template.title = 'Page not found';
            template.description = "We are sorry, but the page you requested doesn't exist.";
            break;
        }
        case 'offline': {
            template.icon = (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: "arrows-exchange-alt-v", size: iconSize[size] });
            template.title = 'No connection';
            template.description = 'Please check our internet connection and try again.';
            break;
        }
    }
    if (!hideIcon && (icon || template.icon)) {
        output.icon = ((0, jsx_runtime_1.jsx)(Box_1.Box, { align: "center", flexBox: true, justify: "center", children: icon ? (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: icon, size: iconSize[size] }) : template.icon }));
    }
    if (title !== null && (title || template.title)) {
        const content = title || template.title;
        switch (size) {
            case 'sm': {
                output.title = (0, jsx_runtime_1.jsx)(Headings_1.H3, { mb: 0, children: content });
                break;
            }
            case 'md': {
                output.title = (0, jsx_runtime_1.jsx)(Headings_1.H2, { mb: 0, children: content });
                break;
            }
            default: {
                output.title = (0, jsx_runtime_1.jsx)(Headings_1.H1, { mb: 0, children: content });
                break;
            }
        }
    }
    if (description !== null && (description || template.description)) {
        output.description = ((0, jsx_runtime_1.jsx)(Paragraph_1.Paragraph, { mt: output.title ? 'xxs' : undefined, children: description || template.description }));
    }
    if (children) {
        output.content = ((0, jsx_runtime_1.jsx)(Box_1.Box, { mt: output.title || output.description ? 'xl' : undefined, children: children }));
    }
    return ((0, jsx_runtime_1.jsxs)(exports.StyledNonIdealState, { "data-component-name": "NonIdealState", ...(0, helpers_1.omit)(props, 'title', 'type'), children: [output.icon, (0, jsx_runtime_1.jsxs)(Box_1.Box, { ml: !isVertical ? 'xs' : undefined, mt: isVertical && output.icon ? 'sm' : undefined, children: [output.title, output.description] }), output.content] }));
}
exports.NonIdealState = NonIdealState;
NonIdealState.defaultProps = {
    direction: 'vertical',
    hideIcon: false,
    maxWidth: '600px',
    padding: 'md',
    radius: false,
    shadow: false,
    size: 'md',
    type: 'not-found',
};
