// @gilbarbara/components/lib/ResponsiveMedia.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponsiveMedia = exports.StyledResponsiveMedia = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const react_2 = require("@emotion/react");
const styled_1 = require("@emotion/styled");
const helpers_1 = require("@gilbarbara/helpers");
const helpers_2 = require("./modules/helpers");
const system_1 = require("./modules/system");
exports.StyledResponsiveMedia = (0, styled_1.default)('div', (0, system_1.getStyledOptions)())(props => {
    const { height, maxWidth, width } = props;
    return (0, react_2.css) `
    ${(0, system_1.baseStyles)(props)};
    max-width: ${maxWidth ? (0, helpers_2.px)(maxWidth) : (0, helpers_2.px)(width)};
    overflow: hidden;
    width: 100%;
    ${(0, system_1.marginStyles)(props)};

    > div {
      padding-bottom: calc(${(0, helpers_1.ceil)(height / width)} * 100%);
      position: relative;
      width: 100%;

      > * {
        height: 100%;
        left: 0;
        max-width: 100%;
        position: absolute;
        top: 0;
      }
    }
  `;
});
exports.ResponsiveMedia = (0, react_1.forwardRef)((props, ref) => {
    const { children, ...rest } = props;
    return ((0, jsx_runtime_1.jsx)(exports.StyledResponsiveMedia, { ref: ref, "data-component-name": "ResponsiveMedia", ...rest, children: (0, jsx_runtime_1.jsx)("div", { children: children }) }));
});
