// @gilbarbara/components/lib/StatusIndicator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatusIndicator = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("@emotion/react");
const styled_1 = require("@emotion/styled");
const helpers_1 = require("./modules/helpers");
const system_1 = require("./modules/system");
const StyledStatusIndicator = (0, styled_1.default)('div', (0, system_1.getStyledOptions)())(props => {
    const { centerShade = 'lighter', ratio = 0.7, shade, size = 24, variant = 'green' } = props;
    const { variants } = (0, helpers_1.getTheme)(props);
    const { bg } = (0, helpers_1.getColorVariant)(variant, shade, variants);
    const { bg: centerBg } = (0, helpers_1.getColorVariant)(variant, centerShade, variants);
    const innerSize = size * ratio < size ? size * ratio : size;
    return (0, react_1.css) `
    align-items: center;
    background-color: ${bg};
    border-radius: 50%;
    display: inline-flex;
    height: ${(0, helpers_1.px)(size)};
    justify-content: center;
    line-height: 1;
    width: ${(0, helpers_1.px)(size)};
    ${(0, system_1.marginStyles)(props)};

    &:before {
      background-color: ${centerBg};
      border-radius: 50%;
      content: '';
      display: block;
      height: ${(0, helpers_1.px)(innerSize)};
      position: absolute;
      width: ${(0, helpers_1.px)(innerSize)};
    }
  `;
});
function StatusIndicator(props) {
    return (0, jsx_runtime_1.jsx)(StyledStatusIndicator, { "data-component-name": "StatusIndicator", ...props });
}
exports.StatusIndicator = StatusIndicator;
StatusIndicator.defaultProps = {
    centerShade: 'lighter',
    ratio: 0.7,
    shade: 'mid',
    size: 24,
    variant: 'green',
};
