// @gilbarbara/components/lib/DatePicker/Input.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatePickerInput = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_use_1 = require("react-use");
const react_1 = require("@emotion/react");
const styled_1 = require("@emotion/styled");
const helpers_1 = require("@gilbarbara/helpers");
const is_lite_1 = require("is-lite");
const Range_1 = require("./Range");
const Single_1 = require("./Single");
const utils_1 = require("./utils");
const Box_1 = require("../Box");
const ClickOutside_1 = require("../ClickOutside");
const Icon_1 = require("../Icon");
const helpers_2 = require("../modules/helpers");
const system_1 = require("../modules/system");
const Text_1 = require("../Text");
const StyledButton = (0, styled_1.default)('div', (0, system_1.getStyledOptions)())(props => {
    const { borderless, isFilled, large, variant = 'primary', width } = props;
    const { darkColor, grayDark, grayDarker, grayMid, inputHeight, lightColor, radius, spacing, variants, white, } = (0, helpers_2.getTheme)(props);
    const darkMode = (0, system_1.isDarkMode)(props);
    const { bg } = (0, helpers_2.getColorVariant)(variant, 'mid', variants);
    let borderColor = darkMode ? grayDark : grayMid;
    let textColor = grayMid;
    if (isFilled) {
        borderColor = bg;
        textColor = darkMode ? lightColor : darkColor;
    }
    const styles = borderless
        ? (0, react_1.css) `
        border: 0 !important;
        border-bottom: 1px solid ${borderColor} !important;
        border-radius: 0 !important;
        padding: ${spacing.xxs} 0 !important;
      `
        : (0, react_1.css) `
        border: 1px solid ${borderColor};
        border-radius: ${radius.xs};
        padding: 0 ${spacing.md} !important;
      `;
    return (0, react_1.css) `
    align-items: center;
    background-color: ${darkMode ? grayDarker : white};
    color: ${textColor};
    cursor: pointer;
    display: flex;
    justify-content: space-between;
    min-height: ${large ? inputHeight.large : inputHeight.normal};
    min-width: 240px;
    width: ${width ? (0, helpers_2.px)(width) : 'auto'};
    ${styles};
  `;
});
const StyledContent = (0, styled_1.default)('div', (0, system_1.getStyledOptions)())(props => {
    const { isActive, position, wide } = props;
    const { grayDarker, radius, shadow, spacing, white } = (0, helpers_2.getTheme)(props);
    const darkMode = (0, system_1.isDarkMode)(props);
    let left = position === 'left' ? 0 : 'auto';
    let translateX = '';
    if (position === 'center') {
        left = '50%';
        translateX = ' translateX(-50%)';
    }
    return (0, react_1.css) `
    background-color: ${darkMode ? grayDarker : white};
    border-radius: ${radius.xxs};
    box-shadow: ${shadow.mid};
    display: flex;
    flex-direction: column;
    justify-content: center;
    left: ${left};
    margin-top: ${spacing.xs};
    min-width: ${(0, helpers_2.px)(wide ? 600 : 300)};
    overflow-y: auto;
    padding: ${spacing.md};
    position: absolute;
    right: ${position === 'right' ? 0 : 'auto'};
    top: 100%;
    transform-origin: top;
    transform: ${`scaleY(0)${translateX}`};
    transition: transform 0.3s;
    z-index: 100;

    ${isActive &&
        (0, react_1.css) `
      transform: ${`scaleY(1)${translateX}`};
    `}
  `;
});
function DatePickerInput(props) {
    const { borderless, large, onSelect, open, placeholder, position = 'right', separator = ' — ', showRange = false, showRangeApply, width, ...rest } = props;
    const [{ isActive, isFilled, selected }, setState] = (0, react_use_1.useSetState)({
        isActive: open || false,
        isFilled: false,
        selected: showRange ? [undefined, undefined] : '',
    });
    const toggle = () => {
        if (is_lite_1.default.boolean(open)) {
            return;
        }
        setState(s => ({ isActive: !s.isActive }));
    };
    const handleApply = (isoDate) => {
        if (onSelect) {
            onSelect(isoDate);
        }
        toggle();
    };
    const handleSelect = (isoDate) => {
        setState({
            isFilled: is_lite_1.default.array(isoDate) ? isoDate.some(Boolean) : !!isoDate,
            selected: isoDate,
        });
        if (onSelect && !showRangeApply) {
            onSelect(isoDate);
        }
        if (!showRangeApply &&
            ((is_lite_1.default.array(isoDate) && isoDate.every(Boolean)) || (!is_lite_1.default.array(isoDate) && isoDate))) {
            toggle();
        }
    };
    const picker = showRange ? ((0, jsx_runtime_1.jsx)(Range_1.DatePickerRange, { ...rest, onApply: handleApply, onSelect: handleSelect, showApply: showRangeApply })) : ((0, jsx_runtime_1.jsx)(Single_1.DatePicker, { ...rest, onSelect: handleSelect }));
    let title = showRange ? 'Select a date range' : 'Select a date';
    if (placeholder) {
        title = placeholder;
    }
    if (is_lite_1.default.array(selected) && selected.some(Boolean)) {
        const dates = selected.reduce((acc, d, index) => {
            acc.push(d ? (0, helpers_1.formatDateLocale)(d) : '???');
            if (index === 0) {
                acc.push(separator);
            }
            return acc;
        }, []);
        title = (0, jsx_runtime_1.jsx)(Text_1.Text, { children: dates.map(d => d) });
    }
    else if (!is_lite_1.default.array(selected) && selected) {
        title = (0, helpers_1.formatDateLocale)(selected);
    }
    const numberOfMonths = (0, utils_1.getNumberOfMonths)(rest.fromDate, rest.toDate);
    return ((0, jsx_runtime_1.jsx)(Box_1.Box, { "data-component-name": "DatePickerInput", position: "relative", children: (0, jsx_runtime_1.jsxs)(ClickOutside_1.ClickOutside, { active: isActive, onClick: toggle, children: [(0, jsx_runtime_1.jsxs)(StyledButton, { "data-component-name": "DatePickerInputButton", isFilled: isFilled, onClick: toggle, ...(0, helpers_1.omit)(props, 'onSelect'), children: [title, (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: "calendar" })] }), (0, jsx_runtime_1.jsx)(StyledContent, { "data-component-name": "DatePickerInputContent", isActive: isActive, position: position, wide: showRange && numberOfMonths > 1, children: picker })] }) }));
}
exports.DatePickerInput = DatePickerInput;
DatePickerInput.defaultProps = {
    ...utils_1.defaultProps,
    borderless: false,
    formatLocale: 'en-US',
    large: false,
    position: 'right',
    separator: ' — ',
    showRange: false,
    showRangeApply: false,
    variant: 'primary',
    width: 'auto',
};
