// @gilbarbara/components/lib/Headings/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.H6 = exports.H5 = exports.H4 = exports.H3 = exports.H2 = exports.H1 = exports.Jumbo = void 0;
var Jumbo_1 = require("./Jumbo");
Object.defineProperty(exports, "Jumbo", { enumerable: true, get: function () { return Jumbo_1.Jumbo; } });
var H1_1 = require("./H1");
Object.defineProperty(exports, "H1", { enumerable: true, get: function () { return H1_1.H1; } });
var H2_1 = require("./H2");
Object.defineProperty(exports, "H2", { enumerable: true, get: function () { return H2_1.H2; } });
var H3_1 = require("./H3");
Object.defineProperty(exports, "H3", { enumerable: true, get: function () { return H3_1.H3; } });
var H4_1 = require("./H4");
Object.defineProperty(exports, "H4", { enumerable: true, get: function () { return H4_1.H4; } });
var H5_1 = require("./H5");
Object.defineProperty(exports, "H5", { enumerable: true, get: function () { return H5_1.H5; } });
var H6_1 = require("./H6");
Object.defineProperty(exports, "H6", { enumerable: true, get: function () { return H6_1.H6; } });
