// @gilbarbara/components/lib/Loader/Grow.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("@emotion/react");
const styled_1 = require("@emotion/styled");
const helpers_1 = require("../modules/helpers");
const system_1 = require("../modules/system");
const grow = ({ size = 32 }) => (0, react_1.keyframes) `
  0% {
    height: 0;
    width: 0;
  }

  30% {
    border-width: ${(0, helpers_1.px)(size && size / 2.5)};
    opacity: 1;
  }

  100% {
    border-width: 0;
    height: ${(0, helpers_1.px)(size)};
    opacity: 0;
    width: ${(0, helpers_1.px)(size)};
  }
`;
const StyledLoaderGrow = (0, styled_1.default)('div', (0, system_1.getStyledOptions)())(props => {
    const { block, color, shade, size = 32, variant } = props;
    const { darkColor, lightColor, spacing, variants } = (0, helpers_1.getTheme)(props);
    const darkMode = (0, system_1.isDarkMode)(props);
    let variantColor = darkMode ? lightColor : darkColor;
    if (variant) {
        variantColor = (0, helpers_1.getColorVariant)(variant, shade, variants).bg;
    }
    return (0, react_1.css) `
    display: ${block ? 'flex' : 'inline-flex'};
    height: ${(0, helpers_1.px)(size)};
    margin: ${block ? spacing.lg : 0} auto;
    position: relative;
    width: ${(0, helpers_1.px)(size)};

    > div {
      animation: ${grow(props)} 1.15s infinite cubic-bezier(0.2, 0.6, 0.36, 1);
      border: 0 solid ${color || variantColor};
      border-radius: 50%;
      height: 0;
      left: 50%;
      position: absolute;
      top: 50%;
      transform: translate(-50%, -50%);
      width: 0;
    }
  `;
});
function LoaderGrow(props) {
    return ((0, jsx_runtime_1.jsx)(StyledLoaderGrow, { "data-component-name": "LoaderGrow", ...props, children: (0, jsx_runtime_1.jsx)("div", {}) }));
}
exports.default = LoaderGrow;
