// @gilbarbara/components/lib/Loader/Pulse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("@emotion/react");
const styled_1 = require("@emotion/styled");
const helpers_1 = require("../modules/helpers");
const system_1 = require("../modules/system");
const pulse = ({ size = 32 }) => (0, react_1.keyframes) `
  0% {
    height: 0;
    left: ${(0, helpers_1.px)(size / 2)};
    opacity: 1;
    top: ${(0, helpers_1.px)(size / 2)};
    width: 0;
  }

  100% {
    height: ${(0, helpers_1.px)(size)};
    left: 0;
    opacity: 0;
    top: 0;
    width: ${(0, helpers_1.px)(size)};
  }
`;
const StyledLoaderPulse = (0, styled_1.default)('div', (0, system_1.getStyledOptions)())(props => {
    const { block, color, shade, size = 32, variant } = props;
    const { darkColor, lightColor, spacing, variants } = (0, helpers_1.getTheme)(props);
    const darkMode = (0, system_1.isDarkMode)(props);
    let variantColor = darkMode ? lightColor : darkColor;
    if (variant) {
        variantColor = (0, helpers_1.getColorVariant)(variant, shade, variants).bg;
    }
    return (0, react_1.css) `
    display: ${block ? 'flex' : 'inline-flex'};
    height: ${(0, helpers_1.px)(size)};
    margin: ${block ? spacing.lg : 0} auto;
    position: relative;
    width: ${(0, helpers_1.px)(size)};

    > div {
      animation: ${pulse(props)} 1.2s cubic-bezier(0, 0.2, 0.8, 1) infinite;
      border: ${(0, helpers_1.px)(Math.round(size / 16))} solid ${color || variantColor};
      border-radius: 50%;
      height: 0;
      left: ${(0, helpers_1.px)(size / 2)};
      opacity: 1;
      position: absolute;
      top: ${(0, helpers_1.px)(size / 2)};
      transform-origin: center;
      width: 0;
    }

    > div:nth-of-type(2) {
      animation-delay: -0.5s;
    }
  `;
});
function LoaderPulse(props) {
    return ((0, jsx_runtime_1.jsxs)(StyledLoaderPulse, { "data-component-name": "LoaderPulse", ...props, children: [(0, jsx_runtime_1.jsx)("div", {}), (0, jsx_runtime_1.jsx)("div", {})] }));
}
exports.default = LoaderPulse;
