// @gilbarbara/components/esm/ButtonSplit.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useCallback, useState } from 'react';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { ButtonBase } from './ButtonBase';
import { Icon } from './Icon';
import { Menu } from './Menu';
import { getTheme } from './modules/helpers';
import { backgroundStyles, getStyledOptions } from './modules/system';
export const StyledButtonSplit = styled('div', getStyledOptions())(props => {
    const { block, disabled, invert, size = 'md' } = props;
    const { button, grayLight, grayMid, spacing } = getTheme(props);
    const { borderRadius, fontSize, fontWeight, height, lineHeight, padding } = button;
    const buttonPadding = `${padding[size][0]} ${padding[size][1]}`;
    const styles = backgroundStyles(props);
    if (disabled) {
        styles.backgroundColor = grayLight;
        styles.borderColor = grayLight;
        styles.color = grayMid;
    }
    return css `
    display: inline-flex;
    min-height: ${height[size]};
    min-width: ${height[size]};
    width: ${block ? '100%' : 'auto'};

    > [data-component-name='ButtonBase'] {
      ${styles};
      border-bottom-left-radius: ${borderRadius[size]};
      border-right: 0;
      border-top-left-radius: ${borderRadius[size]};
      display: flex;
      flex: 1;
      font-size: ${fontSize[size]};
      font-weight: ${fontWeight};
      justify-content: center;
      line-height: ${lineHeight[size]};
      opacity: 1;
      padding: ${buttonPadding};
    }

    > [data-component-name='Menu'] {
      ${styles};
      align-items: center;
      border-left: ${invert ? styles.border : `1px solid ${styles.color}`};
      border-top-right-radius: ${borderRadius[size]};
      border-bottom-right-radius: ${borderRadius[size]};
      display: flex;
    }

    [data-component-name='MenuButton'] {
      height: 100%;
      opacity: 1;
      padding: 0 ${spacing.xs};
      width: 100%;
    }
  `;
});
export function ButtonSplit(props) {
    const { busy, children, label, onClick, onToggle, position, ...rest } = props;
    const { disabled, shade, variant } = rest;
    const [active, setActive] = useState(false);
    const handleToggle = useCallback((status) => {
        setActive(status);
        if (onToggle) {
            onToggle(status);
        }
    }, [onToggle]);
    const buttonProps = {};
    const wrapperProps = {};
    Object.entries(rest).forEach(([key, value]) => {
        if (key.startsWith('data-')) {
            buttonProps[key] = value;
        }
        else {
            wrapperProps[key] = value;
        }
    });
    return (_jsxs(StyledButtonSplit, { "data-component-name": "ButtonSplit", ...wrapperProps, children: [_jsx(ButtonBase, { busy: busy, disabled: disabled, onClick: onClick, ...buttonProps, children: label }), _jsx(Menu, { component: _jsx(Icon, { name: active ? 'chevron-up' : 'chevron-down', size: 24 }), disabled: disabled || busy, onToggle: handleToggle, position: position, shade: shade, variant: variant, children: children })] }));
}
ButtonSplit.defaultProps = {
    block: false,
    busy: false,
    disabled: false,
    invert: false,
    position: 'bottom-right',
    shade: 'mid',
    size: 'md',
    variant: 'primary',
};
export { MenuDivider as ButtonSplitDivider, MenuItem as ButtonSplitItem } from './Menu';
