// @gilbarbara/components/esm/CopyToClipboard.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useEffect, useRef, useState } from 'react';
import { css, useTheme } from '@emotion/react';
import styled from '@emotion/styled';
import { Icon } from './Icon';
import { animateIcon, fadeInOut } from './modules/animations';
import copy from './modules/copy-text-to-clipboard';
import { getTheme } from './modules/helpers';
import { baseStyles, getStyledOptions, marginStyles } from './modules/system';
import { Tooltip } from './Tooltip';
const StyledCopyToClipboard = styled('span', getStyledOptions())(props => {
    return css `
    ${baseStyles(props)};
    cursor: pointer;
    display: inline-flex;
    line-height: 1;
    position: relative;
    ${marginStyles(props)};
  `;
});
const StyledIcon = styled(Icon) `
  transition: transform 0.6s;

  &.will-animate {
    animation: ${fadeInOut} 0.6s ease-out 1 forwards;
  }

  &.is-animating {
    transform: scale(4);
  }
`;
export function CopyToClipboard(props) {
    const { disableAnimation = false, icon = 'copy', onCopy, size = 16, text, tooltipCopiedText = 'Copied!', tooltipText = 'Copy', ...rest } = props;
    const [content, setContent] = useState(tooltipText);
    const isActive = useRef(false);
    const theme = getTheme({ theme: useTheme() });
    useEffect(() => {
        isActive.current = true;
        return () => {
            isActive.current = false;
        };
    }, []);
    const handleClick = (event) => {
        if (!disableAnimation) {
            animateIcon(event.currentTarget, 'primary', theme);
        }
        copy(text);
        setContent(tooltipCopiedText);
        setTimeout(() => {
            if (isActive.current) {
                setContent(tooltipText);
            }
        }, 2000);
    };
    return (_jsx(StyledCopyToClipboard, { "data-component-name": "CopyToClipboard", onClick: handleClick, ...rest, children: _jsx(Tooltip, { content: content, position: "right", size: "small", children: _jsx(StyledIcon, { name: icon, size: size, title: null }) }) }));
}
CopyToClipboard.defaultProps = {
    disableAnimation: false,
    icon: 'copy',
    size: 16,
    tooltipCopiedText: 'Copied!',
    tooltipText: 'Copy',
};
