// @gilbarbara/components/esm/Dialog.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useCallback } from 'react';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { Button } from './Button';
import { H3 } from './Headings';
import { getTheme, px } from './modules/helpers';
import { borderStyles, getStyledOptions, isDarkMode, paddingStyles, radiusStyles, shadowStyles, } from './modules/system';
import { Paragraph } from './Paragraph';
import { Portal } from './Portal';
import { Spacer } from './Spacer';
const StyledDialog = styled('div', getStyledOptions())(props => {
    const { textAlign = 'left', width = 380 } = props;
    const { black, darkColor, white } = getTheme(props);
    const darkMode = isDarkMode(props);
    return css `
      background-color: ${darkMode ? darkColor : white};
      color: ${darkMode ? white : black};
      max-width: 100%;
      text-align: ${textAlign};
      width: ${px(width)};
      ${borderStyles(props)};
      ${paddingStyles(props)};
      ${radiusStyles(props)};
      ${shadowStyles(props)};
    `;
});
export function Dialog(props) {
    const { buttonCancelText, buttonConfirmText, buttonOrder, closeOnClickOverlay, closeOnEsc = true, content, hideOverlay, isActive, onClickCancel, onClickConfirmation, onClose, onOpen, style, title, variant, zIndex, ...rest } = props;
    const handlePortalClose = useCallback(() => {
        onClickCancel();
        if (onClose) {
            onClose();
        }
    }, [onClickCancel, onClose]);
    const actionButton = (_jsx(Button, { "data-test-id": "confirm", onClick: onClickConfirmation, variant: variant, children: buttonConfirmText }));
    const cancelButton = (_jsx(Button, { "data-test-id": "cancel", invert: true, onClick: onClickCancel, variant: variant, children: buttonCancelText }));
    return (_jsx(Portal, { closeOnClickOverlay: closeOnClickOverlay, closeOnEsc: closeOnEsc, hideOverlay: hideOverlay, isActive: isActive, onClose: handlePortalClose, onOpen: onOpen, zIndex: zIndex, children: _jsxs(StyledDialog, { "data-component-name": "Dialog", style: style, ...rest, children: [title && _jsx(H3, { mb: "sm", children: title }), _jsx(Paragraph, { mb: "xl", children: content }), buttonOrder === 'ltr' ? (_jsxs(Spacer, { distribution: "space-between", children: [cancelButton, actionButton] })) : (_jsxs(Spacer, { distribution: "space-between", children: [actionButton, cancelButton] }))] }) }));
}
Dialog.defaultProps = {
    buttonCancelText: 'Cancel',
    buttonConfirmText: 'Confirm',
    buttonOrder: 'ltr',
    closeOnClickOverlay: false,
    closeOnEsc: false,
    hideOverlay: false,
    padding: 'xl',
    radius: 'lg',
    shadow: 'high',
    textAlign: 'left',
    variant: 'primary',
    width: 380,
};
