// @gilbarbara/components/esm/modules/options.js
import { sortByLocaleCompare } from '@gilbarbara/helpers';
export const iconsPack = [
    { name: 'alarm', categories: ['devices'] },
    { name: 'album', categories: ['multimedia'] },
    { name: 'align-bottom', categories: ['layout'] },
    { name: 'align-center', categories: ['layout'] },
    { name: 'align-left', categories: ['layout'] },
    { name: 'align-middle', categories: ['layout'] },
    { name: 'align-right', categories: ['layout'] },
    { name: 'align-top', categories: ['layout'] },
    { name: 'anchor', categories: ['ui'] },
    { name: 'apple-watch', categories: ['devices'] },
    { name: 'arrange-back', categories: ['layout'] },
    { name: 'arrange-front', categories: ['layout'] },
    { name: 'arrow-align-h', categories: ['arrows'] },
    { name: 'arrow-align-v', categories: ['arrows'] },
    { name: 'arrow-bottom-left-o', categories: ['arrows', 'circle'] },
    { name: 'arrow-bottom-left-r', categories: ['arrows', 'square'] },
    { name: 'arrow-bottom-left', categories: ['arrows'] },
    { name: 'arrow-bottom-right-o', categories: ['arrows', 'circle'] },
    { name: 'arrow-bottom-right-r', categories: ['arrows', 'square'] },
    { name: 'arrow-bottom-right', categories: ['arrows'] },
    { name: 'arrow-down-o', categories: ['arrows', 'circle'] },
    { name: 'arrow-down-r', categories: ['arrows', 'square'] },
    { name: 'arrow-down', categories: ['arrows'] },
    { name: 'arrow-left-o', categories: ['arrows', 'circle'] },
    { name: 'arrow-left-r', categories: ['arrows', 'square'] },
    { name: 'arrow-left', categories: ['arrows'] },
    { name: 'arrow-long-down-c', categories: ['arrows'] },
    { name: 'arrow-long-down-e', categories: ['arrows'] },
    { name: 'arrow-long-down-l', categories: ['arrows'] },
    { name: 'arrow-long-down-r', categories: ['arrows'] },
    { name: 'arrow-long-down', categories: ['arrows'] },
    { name: 'arrow-long-left-c', categories: ['arrows'] },
    { name: 'arrow-long-left-e', categories: ['arrows'] },
    { name: 'arrow-long-left-l', categories: ['arrows'] },
    { name: 'arrow-long-left-r', categories: ['arrows'] },
    { name: 'arrow-long-left', categories: ['arrows'] },
    { name: 'arrow-long-right-c', categories: ['arrows'] },
    { name: 'arrow-long-right-e', categories: ['arrows'] },
    { name: 'arrow-long-right-l', categories: ['arrows'] },
    { name: 'arrow-long-right-r', categories: ['arrows'] },
    { name: 'arrow-long-right', categories: ['arrows'] },
    { name: 'arrow-long-up-c', categories: ['arrows'] },
    { name: 'arrow-long-up-e', categories: ['arrows'] },
    { name: 'arrow-long-up-l', categories: ['arrows'] },
    { name: 'arrow-long-up-r', categories: ['arrows'] },
    { name: 'arrow-long-up', categories: ['arrows'] },
    { name: 'arrow-right-o', categories: ['arrows', 'circle'] },
    { name: 'arrow-right-r', categories: ['arrows', 'square'] },
    { name: 'arrow-right', categories: ['arrows'] },
    { name: 'arrow-top-left-o', categories: ['arrows', 'circle'] },
    { name: 'arrow-top-left-r', categories: ['arrows', 'square'] },
    { name: 'arrow-top-left', categories: ['arrows'] },
    { name: 'arrow-top-right-o', categories: ['arrows', 'circle'] },
    { name: 'arrow-top-right-r', categories: ['arrows', 'square'] },
    { name: 'arrow-top-right', categories: ['arrows'] },
    { name: 'arrow-up-o', categories: ['arrows', 'circle'] },
    { name: 'arrow-up-r', categories: ['arrows', 'square'] },
    { name: 'arrow-up', categories: ['arrows'] },
    { name: 'arrows-breake-h', categories: ['arrows'] },
    { name: 'arrows-breake-v', categories: ['arrows'] },
    { name: 'arrows-exchange-alt-v', categories: ['arrows'] },
    { name: 'arrows-exchange-alt', categories: ['arrows'] },
    { name: 'arrows-exchange-v', categories: ['arrows'] },
    { name: 'arrows-exchange', categories: ['arrows'] },
    { name: 'arrows-expand-down-left', categories: ['arrows'] },
    { name: 'arrows-expand-down-right', categories: ['arrows'] },
    { name: 'arrows-expand-left-alt', categories: ['arrows'] },
    { name: 'arrows-expand-left', categories: ['arrows'] },
    { name: 'arrows-expand-right-alt', categories: ['arrows'] },
    { name: 'arrows-expand-right', categories: ['arrows'] },
    { name: 'arrows-expand-up-left', categories: ['arrows'] },
    { name: 'arrows-expand-up-right', categories: ['arrows'] },
    { name: 'arrows-h-alt', categories: ['arrows'] },
    { name: 'arrows-h', categories: ['arrows'] },
    { name: 'arrows-merge-alt-h', categories: ['arrows'] },
    { name: 'arrows-merge-alt-v', categories: ['arrows'] },
    { name: 'arrows-scroll-h', categories: ['arrows'] },
    { name: 'arrows-scroll-v', categories: ['arrows'] },
    { name: 'arrows-shrink-h', categories: ['arrows'] },
    { name: 'arrows-shrink-v', categories: ['arrows'] },
    { name: 'arrows-v-alt', categories: ['arrows'] },
    { name: 'arrows-v', categories: ['arrows'] },
    { name: 'assign', categories: ['ui'] },
    { name: 'asterisk', categories: ['misc'] },
    { name: 'attachment', categories: ['ui'] },
    { name: 'attribution', categories: ['multimedia'] },
    { name: 'awards', categories: ['misc'] },
    { name: 'backspace', categories: ['arrows'] },
    { name: 'band-aid', categories: ['misc'] },
    { name: 'battery', categories: ['devices'] },
    { name: 'battery-empty', categories: ['devices'] },
    { name: 'battery-full', categories: ['devices'] },
    { name: 'bell', categories: ['ui'] },
    { name: 'board', categories: ['ui'] },
    { name: 'bolt', categories: ['misc'] },
    { name: 'bookmark', categories: ['ui'] },
    { name: 'border-all', categories: ['layout'] },
    { name: 'border-bottom', categories: ['layout'] },
    { name: 'border-left', categories: ['layout'] },
    { name: 'border-right', categories: ['layout'] },
    { name: 'border-style-dashed', categories: ['layout'] },
    { name: 'border-style-dotted', categories: ['layout'] },
    { name: 'border-style-solid', categories: ['layout'] },
    { name: 'border-top', categories: ['layout'] },
    { name: 'bot', categories: ['misc'] },
    { name: 'bowl', categories: ['misc'] },
    { name: 'box', categories: ['ui'] },
    { name: 'boy', categories: ['ui'] },
    { name: 'briefcase', categories: ['ui'] },
    { name: 'browse', categories: ['ui'] },
    { name: 'browser', categories: ['ui'] },
    { name: 'brush', categories: ['ui'] },
    { name: 'bulb', categories: ['misc'] },
    { name: 'calculator', categories: ['devices'] },
    { name: 'calendar-dates', categories: ['date'] },
    { name: 'calendar-due', categories: ['date'] },
    { name: 'calendar-next', categories: ['date'] },
    { name: 'calendar-today', categories: ['date'] },
    { name: 'calendar-two', categories: ['date'] },
    { name: 'calendar', categories: ['date'] },
    { name: 'calibrate', categories: ['misc'] },
    { name: 'cap', categories: ['misc'] },
    { name: 'captions', categories: ['ui'] },
    { name: 'card-clubs', categories: ['misc'] },
    { name: 'card-diamonds', categories: ['misc'] },
    { name: 'card-hearts', categories: ['misc'] },
    { name: 'card-spades', categories: ['misc'] },
    { name: 'carousel', categories: ['ui'] },
    { name: 'cast', categories: ['ui'] },
    { name: 'chart', categories: ['ui'] },
    { name: 'check-o', categories: ['ui', 'circle'] },
    { name: 'check-r', categories: ['ui', 'square'] },
    { name: 'chevron-double-down-o', categories: ['arrows', 'circle'] },
    { name: 'chevron-double-down-r', categories: ['arrows', 'square'] },
    { name: 'chevron-double-left-o', categories: ['arrows', 'circle'] },
    { name: 'chevron-double-left-r', categories: ['arrows', 'square'] },
    { name: 'chevron-double-right-o', categories: ['arrows', 'circle'] },
    { name: 'chevron-double-right-r', categories: ['arrows', 'square'] },
    { name: 'chevron-double-up-o', categories: ['arrows', 'circle'] },
    { name: 'chevron-double-up-r', categories: ['arrows', 'square'] },
    { name: 'chevron-down-o', categories: ['arrows', 'circle'] },
    { name: 'chevron-down-r', categories: ['arrows', 'square'] },
    { name: 'chevron-down', categories: ['arrows'] },
    { name: 'chevron-left-o', categories: ['arrows', 'circle'] },
    { name: 'chevron-left-r', categories: ['arrows', 'square'] },
    { name: 'chevron-left', categories: ['arrows'] },
    { name: 'chevron-right-o', categories: ['arrows', 'circle'] },
    { name: 'chevron-right-r', categories: ['arrows', 'square'] },
    { name: 'chevron-right', categories: ['arrows'] },
    { name: 'chevron-up-o', categories: ['arrows', 'circle'] },
    { name: 'chevron-up-r', categories: ['arrows', 'square'] },
    { name: 'chevron-up', categories: ['arrows'] },
    { name: 'clapper-board', categories: ['multimedia'] },
    { name: 'clipboard', categories: ['ui'] },
    { name: 'close-o', categories: ['ui', 'circle'] },
    { name: 'close-r', categories: ['ui', 'square'] },
    { name: 'close', categories: ['ui'] },
    { name: 'cloud', categories: ['ui'] },
    { name: 'code', categories: ['development'] },
    { name: 'code-slash', categories: ['development'] },
    { name: 'coffee', categories: ['misc'] },
    { name: 'collage', categories: ['ui'] },
    { name: 'color-bucket', categories: ['ui'] },
    { name: 'color-picker', categories: ['ui'] },
    { name: 'comment', categories: ['ui'] },
    { name: 'community', categories: ['misc'] },
    { name: 'components', categories: ['ui'] },
    { name: 'compress-left', categories: ['arrows'] },
    { name: 'compress-right', categories: ['arrows'] },
    { name: 'compress-v', categories: ['arrows'] },
    { name: 'compress', categories: ['arrows'] },
    { name: 'controller', categories: ['ui'] },
    { name: 'copy', categories: ['ui'] },
    { name: 'copyright', categories: ['ui'] },
    { name: 'corner-double-down-left', categories: ['arrows'] },
    { name: 'corner-double-down-right', categories: ['arrows'] },
    { name: 'corner-double-left-down', categories: ['arrows'] },
    { name: 'corner-double-left-up', categories: ['arrows'] },
    { name: 'corner-double-right-down', categories: ['arrows'] },
    { name: 'corner-double-right-up', categories: ['arrows'] },
    { name: 'corner-double-up-left', categories: ['arrows'] },
    { name: 'corner-double-up-right', categories: ['arrows'] },
    { name: 'corner-down-left', categories: ['arrows'] },
    { name: 'corner-down-right', categories: ['arrows'] },
    { name: 'corner-left-down', categories: ['arrows'] },
    { name: 'corner-left-up', categories: ['arrows'] },
    { name: 'corner-right-down', categories: ['arrows'] },
    { name: 'corner-right-up', categories: ['arrows'] },
    { name: 'corner-up-left', categories: ['arrows'] },
    { name: 'corner-up-right', categories: ['arrows'] },
    { name: 'credit-card', categories: ['misc'] },
    { name: 'crop', categories: ['ui'] },
    { name: 'crown', categories: ['misc'] },
    { name: 'dark-mode', categories: ['ui'] },
    { name: 'data', categories: ['multimedia'] },
    { name: 'database', categories: ['ui'] },
    { name: 'debug', categories: ['development'] },
    { name: 'desktop', categories: ['ui'] },
    { name: 'details-less', categories: ['layout'] },
    { name: 'details-more', categories: ['layout'] },
    { name: 'dialpad', categories: ['ui'] },
    { name: 'dice-1', categories: ['misc'] },
    { name: 'dice-2', categories: ['misc'] },
    { name: 'dice-3', categories: ['misc'] },
    { name: 'dice-4', categories: ['misc'] },
    { name: 'dice-5', categories: ['misc'] },
    { name: 'dice-6', categories: ['misc'] },
    { name: 'disc', categories: ['multimedia'] },
    { name: 'display-flex', categories: ['layout'] },
    { name: 'display-fullwidth', categories: ['layout'] },
    { name: 'display-grid', categories: ['layout'] },
    { name: 'display-spacing', categories: ['layout'] },
    { name: 'distribute-horizontal', categories: ['layout'] },
    { name: 'distribute-vertical', categories: ['layout'] },
    { name: 'dock-bottom', categories: ['ui'] },
    { name: 'dock-left', categories: ['ui'] },
    { name: 'dock-right', categories: ['ui'] },
    { name: 'dock-window', categories: ['ui'] },
    { name: 'dollar', categories: ['misc'] },
    { name: 'drive', categories: ['devices'] },
    { name: 'drop-invert', categories: ['ui'] },
    { name: 'drop-opacity', categories: ['ui'] },
    { name: 'drop', categories: ['ui'] },
    { name: 'duplicate', categories: ['ui'] },
    { name: 'edit-black-point', categories: ['ui'] },
    { name: 'edit-contrast', categories: ['ui'] },
    { name: 'edit-exposure', categories: ['ui'] },
    { name: 'edit-fade', categories: ['ui'] },
    { name: 'edit-flip-h', categories: ['ui'] },
    { name: 'edit-flip-v', categories: ['ui'] },
    { name: 'edit-highlight', categories: ['ui'] },
    { name: 'edit-markup', categories: ['ui'] },
    { name: 'edit-mask', categories: ['ui'] },
    { name: 'edit-noise', categories: ['ui'] },
    { name: 'edit-shadows', categories: ['ui'] },
    { name: 'edit-straight', categories: ['ui'] },
    { name: 'edit-unmask', categories: ['ui'] },
    { name: 'eject', categories: ['ui'] },
    { name: 'enter', categories: ['ui'] },
    { name: 'erase', categories: ['ui'] },
    { name: 'ereader', categories: ['devices'] },
    { name: 'ethernet', categories: ['ui'] },
    { name: 'euro', categories: ['misc'] },
    { name: 'expand', categories: ['arrows'] },
    { name: 'export', categories: ['ui'] },
    { name: 'extension-add', categories: ['ui'] },
    { name: 'extension-alt', categories: ['ui'] },
    { name: 'extension-remove', categories: ['ui'] },
    { name: 'extension', categories: ['ui'] },
    { name: 'external', categories: ['ui'] },
    { name: 'eye-alt', categories: ['ui'] },
    { name: 'eye', categories: ['ui'] },
    { name: 'feed', categories: ['ui'] },
    { name: 'file-add', categories: ['ui'] },
    { name: 'file-document', categories: ['ui'] },
    { name: 'file-remove', categories: ['ui'] },
    { name: 'file', categories: ['ui'] },
    { name: 'film', categories: ['multimedia'] },
    { name: 'filters', categories: ['ui'] },
    { name: 'flag-alt', categories: ['ui'] },
    { name: 'flag', categories: ['ui'] },
    { name: 'folder-add', categories: ['ui'] },
    { name: 'folder-remove', categories: ['ui'] },
    { name: 'folder', categories: ['ui'] },
    { name: 'font-height', categories: ['typography'] },
    { name: 'font-spacing', categories: ['typography'] },
    { name: 'format-center', categories: ['typography'] },
    { name: 'format-color', categories: ['typography'] },
    { name: 'format-heading', categories: ['typography'] },
    { name: 'format-indent-decrease', categories: ['typography'] },
    { name: 'format-indent-increase', categories: ['typography'] },
    { name: 'format-italic', categories: ['typography'] },
    { name: 'format-justify', categories: ['typography'] },
    { name: 'format-left', categories: ['typography'] },
    { name: 'format-line-height', categories: ['typography'] },
    { name: 'format-right', categories: ['typography'] },
    { name: 'format-separator', categories: ['typography'] },
    { name: 'format-slash', categories: ['typography'] },
    { name: 'format-strike', categories: ['typography'] },
    { name: 'format-text', categories: ['typography'] },
    { name: 'format-underline', categories: ['typography'] },
    { name: 'format-uppercase', categories: ['typography'] },
    { name: 'games', categories: ['devices'] },
    { name: 'gender-female', categories: ['ui'] },
    { name: 'gender-male', categories: ['ui'] },
    { name: 'gift', categories: ['misc'] },
    { name: 'girl', categories: ['ui'] },
    { name: 'git-branch', categories: ['development'] },
    { name: 'git-commit', categories: ['development'] },
    { name: 'git-fork', categories: ['development'] },
    { name: 'git-pull', categories: ['development'] },
    { name: 'glass-alt', categories: ['misc'] },
    { name: 'globe-alt', categories: ['ui'] },
    { name: 'globe', categories: ['ui'] },
    { name: 'gym', categories: ['misc'] },
    { name: 'hashtag', categories: ['ui'] },
    { name: 'headset', categories: ['devices', 'multimedia'] },
    { name: 'heart', categories: ['ui'] },
    { name: 'hello', categories: ['misc'] },
    { name: 'home-alt', categories: ['ui'] },
    { name: 'home-screen', categories: ['devices'] },
    { name: 'home', categories: ['ui'] },
    { name: 'icecream', categories: ['misc'] },
    { name: 'image', categories: ['ui'] },
    { name: 'import', categories: ['ui'] },
    { name: 'inbox', categories: ['ui'] },
    { name: 'infinity', categories: ['misc'] },
    { name: 'inpicture', categories: ['multimedia'] },
    { name: 'insert-after-o', categories: ['ui'] },
    { name: 'insert-after-r', categories: ['ui'] },
    { name: 'insert-after', categories: ['ui'] },
    { name: 'insert-before-o', categories: ['ui'] },
    { name: 'insert-before-r', categories: ['ui'] },
    { name: 'insert-before', categories: ['ui'] },
    { name: 'insights', categories: ['ui'] },
    { name: 'internal', categories: ['ui'] },
    { name: 'key', categories: ['ui'] },
    { name: 'keyboard', categories: ['devices'] },
    { name: 'laptop', categories: ['devices'] },
    { name: 'layout-grid-small', categories: ['layout'] },
    { name: 'layout-grid', categories: ['layout'] },
    { name: 'layout-list', categories: ['layout'] },
    { name: 'layout-pin', categories: ['layout'] },
    { name: 'link', categories: ['ui'] },
    { name: 'list-tree', categories: ['ui'] },
    { name: 'list', categories: ['ui'] },
    { name: 'live-photo', categories: ['multimedia'] },
    { name: 'lock-unlock', categories: ['ui'] },
    { name: 'lock', categories: ['ui'] },
    { name: 'log-in', categories: ['ui'] },
    { name: 'log-off', categories: ['ui'] },
    { name: 'log-out', categories: ['ui'] },
    { name: 'loupe', categories: ['ui'] },
    { name: 'magnet', categories: ['misc'] },
    { name: 'mail-forward', categories: ['ui'] },
    { name: 'mail-open', categories: ['ui'] },
    { name: 'mail-reply', categories: ['ui'] },
    { name: 'mail', categories: ['ui'] },
    { name: 'math-divide', categories: ['ui'] },
    { name: 'math-equal', categories: ['ui'] },
    { name: 'math-minus', categories: ['ui'] },
    { name: 'math-percent', categories: ['ui'] },
    { name: 'math-plus', categories: ['ui'] },
    { name: 'maximize-alt', categories: ['ui'] },
    { name: 'maximize', categories: ['ui'] },
    { name: 'media-live', categories: ['multimedia'] },
    { name: 'media-podcast', categories: ['multimedia'] },
    { name: 'menu-boxed', categories: ['ui'] },
    { name: 'menu-cake', categories: ['ui'] },
    { name: 'menu-cheese', categories: ['ui'] },
    { name: 'menu-grid-o', categories: ['ui'] },
    { name: 'menu-grid-r', categories: ['ui'] },
    { name: 'menu-hotdog', categories: ['ui'] },
    { name: 'menu-left-alt', categories: ['ui'] },
    { name: 'menu-left', categories: ['ui'] },
    { name: 'menu-motion', categories: ['ui'] },
    { name: 'menu-oreos', categories: ['ui'] },
    { name: 'menu-right-alt', categories: ['ui'] },
    { name: 'menu-right', categories: ['ui'] },
    { name: 'menu', categories: ['ui'] },
    { name: 'merge-horizontal', categories: ['ui'] },
    { name: 'merge-vertical', categories: ['ui'] },
    { name: 'mini-player', categories: ['multimedia'] },
    { name: 'minimize-alt', categories: ['ui'] },
    { name: 'minimize', categories: ['ui'] },
    { name: 'modem', categories: ['devices'] },
    { name: 'moon', categories: ['misc'] },
    { name: 'more-alt', categories: ['ui'] },
    { name: 'more-r', categories: ['ui'] },
    { name: 'more-vertical-alt', categories: ['ui'] },
    { name: 'more-vertical-r', categories: ['ui'] },
    { name: 'more-vertical', categories: ['ui'] },
    { name: 'more', categories: ['ui'] },
    { name: 'mouse', categories: ['devices'] },
    { name: 'move-down', categories: ['ui'] },
    { name: 'move-left', categories: ['ui'] },
    { name: 'move-right', categories: ['ui'] },
    { name: 'move-task', categories: ['ui'] },
    { name: 'move-up', categories: ['ui'] },
    { name: 'music', categories: ['multimedia'] },
    { name: 'music-note', categories: ['multimedia'] },
    { name: 'music-speaker', categories: ['multimedia'] },
    { name: 'nametag', categories: ['ui'] },
    { name: 'notes', categories: ['ui'] },
    { name: 'notifications', categories: ['ui'] },
    { name: 'options', categories: ['ui'] },
    { name: 'organisation', categories: ['ui'] },
    { name: 'pacman', categories: ['misc'] },
    { name: 'password', categories: ['ui'] },
    { name: 'path-back', categories: ['layout'] },
    { name: 'path-crop', categories: ['layout'] },
    { name: 'path-divide', categories: ['layout'] },
    { name: 'path-exclude', categories: ['layout'] },
    { name: 'path-front', categories: ['layout'] },
    { name: 'path-intersect', categories: ['layout'] },
    { name: 'path-outline', categories: ['layout'] },
    { name: 'path-trim', categories: ['layout'] },
    { name: 'path-unite', categories: ['layout'] },
    { name: 'pen', categories: ['ui'] },
    { name: 'pentagon-bottom-left', categories: ['misc'] },
    { name: 'pentagon-bottom-right', categories: ['misc'] },
    { name: 'pentagon-down', categories: ['misc'] },
    { name: 'pentagon-left', categories: ['misc'] },
    { name: 'pentagon-right', categories: ['misc'] },
    { name: 'pentagon-top-left', categories: ['misc'] },
    { name: 'pentagon-top-right', categories: ['misc'] },
    { name: 'pentagon-up', categories: ['misc'] },
    { name: 'performance', categories: ['ui'] },
    { name: 'phone', categories: ['devices'] },
    { name: 'photoscan', categories: ['devices'] },
    { name: 'piano', categories: ['multimedia'] },
    { name: 'pill', categories: ['misc'] },
    { name: 'pin-alt', categories: ['ui'] },
    { name: 'pin-bottom', categories: ['ui'] },
    { name: 'pin-top', categories: ['ui'] },
    { name: 'pin', categories: ['ui'] },
    { name: 'play-backwards', categories: ['multimedia'] },
    { name: 'play-button-o', categories: ['multimedia', 'circle'] },
    { name: 'play-button-r', categories: ['multimedia', 'square'] },
    { name: 'play-button', categories: ['multimedia'] },
    { name: 'play-forwards', categories: ['multimedia'] },
    { name: 'play-list-add', categories: ['multimedia'] },
    { name: 'play-list-check', categories: ['multimedia'] },
    { name: 'play-list-remove', categories: ['multimedia'] },
    { name: 'play-list-search', categories: ['multimedia'] },
    { name: 'play-list', categories: ['multimedia'] },
    { name: 'play-pause-o', categories: ['multimedia', 'circle'] },
    { name: 'play-pause-r', categories: ['multimedia', 'square'] },
    { name: 'play-pause', categories: ['multimedia'] },
    { name: 'play-stop-o', categories: ['multimedia', 'circle'] },
    { name: 'play-stop-r', categories: ['multimedia', 'square'] },
    { name: 'play-stop', categories: ['multimedia'] },
    { name: 'play-track-next-o', categories: ['multimedia', 'circle'] },
    { name: 'play-track-next-r', categories: ['multimedia', 'square'] },
    { name: 'play-track-next', categories: ['multimedia'] },
    { name: 'play-track-prev-o', categories: ['multimedia', 'circle'] },
    { name: 'play-track-prev-r', categories: ['multimedia', 'square'] },
    { name: 'play-track-prev', categories: ['multimedia'] },
    { name: 'plug', categories: ['ui'] },
    { name: 'pokemon', categories: ['misc'] },
    { name: 'polaroid', categories: ['misc'] },
    { name: 'poll', categories: ['ui'] },
    { name: 'presentation', categories: ['ui'] },
    { name: 'printer', categories: ['devices'] },
    { name: 'profile', categories: ['ui'] },
    { name: 'pull-clear', categories: ['ui'] },
    { name: 'push-chevron-down-o', categories: ['ui', 'circle'] },
    { name: 'push-chevron-down-r', categories: ['ui', 'square'] },
    { name: 'push-chevron-down', categories: ['ui'] },
    { name: 'push-chevron-left-o', categories: ['ui', 'circle'] },
    { name: 'push-chevron-left-r', categories: ['ui', 'square'] },
    { name: 'push-chevron-left', categories: ['ui'] },
    { name: 'push-chevron-right-o', categories: ['ui', 'circle'] },
    { name: 'push-chevron-right-r', categories: ['ui', 'square'] },
    { name: 'push-chevron-right', categories: ['ui'] },
    { name: 'push-chevron-up-o', categories: ['ui', 'circle'] },
    { name: 'push-chevron-up-r', categories: ['ui', 'square'] },
    { name: 'push-chevron-up', categories: ['ui'] },
    { name: 'push-down', categories: ['ui'] },
    { name: 'push-left', categories: ['ui'] },
    { name: 'push-right', categories: ['ui'] },
    { name: 'push-up', categories: ['ui'] },
    { name: 'qr', categories: ['ui'] },
    { name: 'quote-o', categories: ['ui'] },
    { name: 'quote', categories: ['ui'] },
    { name: 'radio-check', categories: ['ui'] },
    { name: 'radio-checked', categories: ['ui'] },
    { name: 'ratio', categories: ['ui'] },
    { name: 'read', categories: ['misc'] },
    { name: 'readme', categories: ['ui'] },
    { name: 'redo', categories: ['ui'] },
    { name: 'remote', categories: ['ui'] },
    { name: 'remove-r', categories: ['ui', 'square'] },
    { name: 'remove', categories: ['ui'] },
    { name: 'rename', categories: ['ui'] },
    { name: 'reorder', categories: ['ui'] },
    { name: 'repeat', categories: ['ui'] },
    { name: 'ring', categories: ['misc'] },
    { name: 'row-first', categories: ['layout'] },
    { name: 'row-last', categories: ['layout'] },
    { name: 'ruler', categories: ['ui'] },
    { name: 'sand-clock', categories: ['ui'] },
    { name: 'scan', categories: ['ui'] },
    { name: 'screen-mirror', categories: ['ui'] },
    { name: 'screen-shot', categories: ['ui'] },
    { name: 'screen-wide', categories: ['ui'] },
    { name: 'screen', categories: ['ui'] },
    { name: 'scroll-h', categories: ['ui'] },
    { name: 'scroll-v', categories: ['ui'] },
    { name: 'search-found', categories: ['ui'] },
    { name: 'search-loading', categories: ['ui'] },
    { name: 'search', categories: ['ui'] },
    { name: 'select-o', categories: ['ui', 'circle'] },
    { name: 'select-r', categories: ['ui', 'square'] },
    { name: 'select', categories: ['ui'] },
    { name: 'server', categories: ['ui'] },
    { name: 'shape-circle', categories: ['misc'] },
    { name: 'shape-half-circle', categories: ['misc'] },
    { name: 'shape-hexagon', categories: ['misc'] },
    { name: 'shape-rhombus', categories: ['misc'] },
    { name: 'shape-square', categories: ['misc'] },
    { name: 'shape-triangle', categories: ['misc'] },
    { name: 'shape-zigzag', categories: ['misc'] },
    { name: 'share', categories: ['ui'] },
    { name: 'shield', categories: ['misc'] },
    { name: 'shopping-bag', categories: ['ui'] },
    { name: 'shopping-cart', categories: ['ui'] },
    { name: 'shortcut', categories: ['ui'] },
    { name: 'sidebar-open', categories: ['ui'] },
    { name: 'sidebar-right', categories: ['ui'] },
    { name: 'sidebar', categories: ['ui'] },
    { name: 'signal', categories: ['ui'] },
    { name: 'size', categories: ['ui'] },
    { name: 'sketch', categories: ['misc'] },
    { name: 'sleep', categories: ['ui'] },
    { name: 'smart-home-boiler', categories: ['misc'] },
    { name: 'smart-home-cooker', categories: ['misc'] },
    { name: 'smart-home-heat', categories: ['misc'] },
    { name: 'smart-home-light', categories: ['misc'] },
    { name: 'smart-home-refrigerator', categories: ['misc'] },
    { name: 'smart-home-wash-machine', categories: ['misc'] },
    { name: 'smartphone', categories: ['devices'] },
    { name: 'smartphone-chip', categories: ['devices'] },
    { name: 'smartphone-ram', categories: ['devices'] },
    { name: 'smartphone-shake', categories: ['devices'] },
    { name: 'smile-mouth-open', categories: ['ui'] },
    { name: 'smile-neutral', categories: ['ui'] },
    { name: 'smile-no-mouth', categories: ['ui'] },
    { name: 'smile-none', categories: ['ui'] },
    { name: 'smile-sad', categories: ['ui'] },
    { name: 'smile-upside', categories: ['ui'] },
    { name: 'smile', categories: ['ui'] },
    { name: 'software-download', categories: ['ui'] },
    { name: 'software-upload', categories: ['ui'] },
    { name: 'space-between-v', categories: ['layout'] },
    { name: 'space-between', categories: ['layout'] },
    { name: 'spinner-alt', categories: ['loaders'] },
    { name: 'spinner-two-alt', categories: ['loaders'] },
    { name: 'spinner-two', categories: ['loaders'] },
    { name: 'spinner', categories: ['loaders'] },
    { name: 'stack', categories: ['layout'] },
    { name: 'stopwatch', categories: ['devices'] },
    { name: 'stories', categories: ['ui'] },
    { name: 'style', categories: ['ui'] },
    { name: 'sun', categories: ['ui'] },
    { name: 'support', categories: ['ui'] },
    { name: 'swap-vertical', categories: ['ui'] },
    { name: 'swap', categories: ['ui'] },
    { name: 'sync', categories: ['ui'] },
    { name: 'tab', categories: ['ui'] },
    { name: 'tap-double', categories: ['ui'] },
    { name: 'tap-single', categories: ['ui'] },
    { name: 'template', categories: ['ui'] },
    { name: 'tennis', categories: ['misc'] },
    { name: 'terminal', categories: ['development'] },
    { name: 'terrain', categories: ['misc'] },
    { name: 'thermometer', categories: ['misc'] },
    { name: 'thermostat', categories: ['misc'] },
    { name: 'tikcode', categories: ['misc'] },
    { name: 'time', categories: ['ui'] },
    { name: 'timelapse', categories: ['ui'] },
    { name: 'timer', categories: ['ui'] },
    { name: 'today', categories: ['ui'] },
    { name: 'toggle-off', categories: ['ui'] },
    { name: 'toggle-on', categories: ['ui'] },
    { name: 'toggle-square-off', categories: ['ui'] },
    { name: 'toggle-square', categories: ['ui'] },
    { name: 'toolbar-bottom', categories: ['ui'] },
    { name: 'toolbar-left', categories: ['ui'] },
    { name: 'toolbar-right', categories: ['ui'] },
    { name: 'toolbar-top', categories: ['ui'] },
    { name: 'toolbox', categories: ['ui'] },
    { name: 'touchpad', categories: ['devices'] },
    { name: 'track', categories: ['ui'] },
    { name: 'transcript', categories: ['ui'] },
    { name: 'trash-empty', categories: ['ui'] },
    { name: 'trash', categories: ['ui'] },
    { name: 'tree', categories: ['misc'] },
    { name: 'trees', categories: ['misc'] },
    { name: 'trending-down', categories: ['ui'] },
    { name: 'trending', categories: ['ui'] },
    { name: 'tv', categories: ['devices'] },
    { name: 'ui-kit', categories: ['misc'] },
    { name: 'umbrella', categories: ['misc'] },
    { name: 'undo', categories: ['ui'] },
    { name: 'usb', categories: ['devices'] },
    { name: 'usb-c', categories: ['devices'] },
    { name: 'user-add', categories: ['ui'] },
    { name: 'user-list', categories: ['ui'] },
    { name: 'user-remove', categories: ['ui'] },
    { name: 'user', categories: ['ui'] },
    { name: 'view-cols', categories: ['layout'] },
    { name: 'view-comfortable', categories: ['layout'] },
    { name: 'view-day', categories: ['layout'] },
    { name: 'view-grid', categories: ['layout'] },
    { name: 'view-list', categories: ['layout'] },
    { name: 'view-month', categories: ['layout'] },
    { name: 'view-split', categories: ['layout'] },
    { name: 'voicemail-o', categories: ['ui', 'circle'] },
    { name: 'voicemail-r', categories: ['ui'] },
    { name: 'voicemail', categories: ['ui'] },
    { name: 'website', categories: ['ui'] },
    { name: 'work-alt', categories: ['ui'] },
    { name: 'yinyang', categories: ['misc'] },
    { name: 'zoom-in', categories: ['ui'] },
    { name: 'zoom-out', categories: ['ui'] },
];
export const iconsCustom = [
    { name: 'accessibility', categories: ['ui', 'circle'] },
    { name: 'add', categories: ['ui', 'circle'] },
    { name: 'add-r', categories: ['ui', 'square'] },
    { name: 'airplane', categories: ['misc'] },
    { name: 'angular', categories: ['development'] },
    { name: 'apple', categories: ['brand'] },
    { name: 'atlassian', categories: ['development'] },
    { name: 'bamboo', categories: ['development'] },
    { name: 'bitbucket', categories: ['development'] },
    { name: 'block', categories: ['ui', 'circle'] },
    { name: 'bullseye', categories: ['misc'] },
    { name: 'bullseye-arrow', categories: ['misc'] },
    { name: 'c-plus-plus', categories: ['development'] },
    { name: 'camera', categories: ['devices', 'multimedia'] },
    { name: 'camera-alt', categories: ['devices', 'multimedia'] },
    { name: 'categories', categories: ['ui'] },
    { name: 'check', categories: ['ui'] },
    { name: 'chevron-double-down', categories: ['arrows'] },
    { name: 'chevron-double-left', categories: ['arrows'] },
    { name: 'chevron-double-right', categories: ['arrows'] },
    { name: 'chevron-double-up', categories: ['arrows'] },
    { name: 'circle-ci', categories: ['development'] },
    { name: 'code-climate', categories: ['development'] },
    { name: 'codepen', categories: ['development'] },
    { name: 'codesandbox', categories: ['development'] },
    { name: 'cog', categories: ['ui'] },
    { name: 'cross', categories: ['misc'] },
    { name: 'danger', categories: ['ui', 'circle'] },
    { name: 'digital-ocean', categories: ['development'] },
    { name: 'discord', categories: ['brand'] },
    { name: 'dribble', categories: ['brand'] },
    { name: 'dropbox', categories: ['brand'] },
    { name: 'facebook', categories: ['brand'] },
    { name: 'figma', categories: ['brand'] },
    { name: 'format-bold', categories: ['typography'] },
    { name: 'framer', categories: ['brand'] },
    { name: 'ghost', categories: ['misc'] },
    { name: 'git', categories: ['development'] },
    { name: 'github', categories: ['development'] },
    { name: 'github-actions', categories: ['development'] },
    { name: 'github-copilot', categories: ['development'] },
    { name: 'gitlab', categories: ['development'] },
    { name: 'gitter', categories: ['development'] },
    { name: 'glass', categories: ['misc'] },
    { name: 'google', categories: ['brand'] },
    { name: 'google-calendar', categories: ['brand'] },
    { name: 'google-drive', categories: ['brand'] },
    { name: 'google-meet', categories: ['brand'] },
    { name: 'heart-filled', categories: ['ui'] },
    { name: 'info', categories: ['ui', 'circle'] },
    { name: 'instagram', categories: ['brand'] },
    { name: 'invision', categories: ['brand'] },
    { name: 'javascript', categories: ['development'] },
    { name: 'jenkins', categories: ['development'] },
    { name: 'jest', categories: ['development'] },
    { name: 'jsfiddle', categories: ['development'] },
    { name: 'keyhole', categories: ['ui', 'circle'] },
    { name: 'kotlin', categories: ['development'] },
    { name: 'linkedin', categories: ['brand'] },
    { name: 'loadbar', categories: ['loaders'] },
    { name: 'loadbar-alt', categories: ['loaders'] },
    { name: 'loadbar-doc', categories: ['loaders'] },
    { name: 'loadbar-sound', categories: ['loaders'] },
    { name: 'loom', categories: ['brand'] },
    { name: 'markdown', categories: ['development'] },
    { name: 'menu-close', categories: ['ui'] },
    { name: 'menu-open', categories: ['ui'] },
    { name: 'menu-round', categories: ['ui', 'circle'] },
    { name: 'messenger', categories: ['brand'] },
    { name: 'meta', categories: ['brand'] },
    { name: 'mic', categories: ['devices', 'multimedia'] },
    { name: 'mic-muted', categories: ['devices', 'multimedia'] },
    { name: 'microsoft', categories: ['brand'] },
    { name: 'microsoft-teams', categories: ['brand'] },
    { name: 'microsoft-windows', categories: ['brand'] },
    { name: 'money-sign', categories: ['misc'] },
    { name: 'more-o', categories: ['ui', 'circle'] },
    { name: 'more-vertical-o', categories: ['ui', 'circle'] },
    { name: 'next.js', categories: ['development'] },
    { name: 'node.js', categories: ['development'] },
    { name: 'npm', categories: ['development'] },
    { name: 'password-alt', categories: ['ui'] },
    { name: 'patreon', categories: ['brand'] },
    { name: 'paypal', categories: ['brand'] },
    { name: 'product-hunt', categories: ['brand'] },
    { name: 'python', categories: ['development'] },
    { name: 'react', categories: ['development'] },
    { name: 'record', categories: ['multimedia', 'circle'] },
    { name: 'redux', categories: ['development'] },
    { name: 'rocket-chat', categories: ['brand'] },
    { name: 'rust', categories: ['development'] },
    { name: 'send', categories: ['ui'] },
    { name: 'send-alt', categories: ['ui'] },
    { name: 'serverless', categories: ['development'] },
    { name: 'skype', categories: ['brand'] },
    { name: 'slack', categories: ['brand'] },
    { name: 'sort', categories: ['ui'] },
    { name: 'sort-asc', categories: ['ui'] },
    { name: 'sort-desc', categories: ['ui'] },
    { name: 'spotify', categories: ['brand'] },
    { name: 'square', categories: ['brand'] },
    { name: 'star', categories: ['ui'] },
    { name: 'svelte', categories: ['development'] },
    { name: 'swift', categories: ['development'] },
    { name: 'tag', categories: ['ui'] },
    { name: 'tags', categories: ['ui'] },
    { name: 'thumbs-down', categories: ['ui'] },
    { name: 'thumbs-up', categories: ['ui'] },
    { name: 'tidal', categories: ['brand'] },
    { name: 'travis-ci', categories: ['development'] },
    { name: 'trello', categories: ['brand'] },
    { name: 'trophy', categories: ['misc'] },
    { name: 'tumblr', categories: ['brand'] },
    { name: 'twilio', categories: ['development'] },
    { name: 'twitch', categories: ['brand'] },
    { name: 'twitter', categories: ['brand'] },
    { name: 'typescript', categories: ['development'] },
    { name: 'unavailable', categories: ['ui', 'circle'] },
    { name: 'unblock', categories: ['ui', 'circle'] },
    { name: 'vector', categories: ['ui'] },
    { name: 'vercel', categories: ['development'] },
    { name: 'video-camera', categories: ['devices', 'multimedia'] },
    { name: 'vimeo', categories: ['brand'] },
    { name: 'vinyl', categories: ['multimedia'] },
    { name: 'vscode', categories: ['development'] },
    { name: 'vue', categories: ['development'] },
    { name: 'volume-low', categories: ['multimedia'] },
    { name: 'volume-mid', categories: ['multimedia'] },
    { name: 'volume-high', categories: ['multimedia'] },
    { name: 'webcam', categories: ['devices', 'multimedia'] },
    { name: 'webstorm', categories: ['development'] },
    { name: 'whatsapp', categories: ['brand'] },
    { name: 'wheelchair', categories: ['ui'] },
    { name: 'workplace', categories: ['brand'] },
    { name: 'youtube', categories: ['brand'] },
    { name: 'zoom', categories: ['brand'] },
];
export const icons = [...iconsCustom, ...iconsPack].sort(sortByLocaleCompare('name'));
export const inputTypes = [
    'color',
    'date',
    'email',
    'file',
    'hidden',
    'number',
    'password',
    'search',
    'tel',
    'text',
];
export const sizes = ['sm', 'md', 'lg'];
export const textSizes = ['small', 'mid', 'regular', 'large'];
