// @gilbarbara/components/lib/DataTable/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataTable = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const react_innertext_1 = require("react-innertext");
const react_use_1 = require("react-use");
const react_2 = require("@emotion/react");
const Body_1 = require("./Body");
const Head_1 = require("./Head");
const Box_1 = require("../Box");
const animations_1 = require("../modules/animations");
const helpers_1 = require("../modules/helpers");
const Pagination_1 = require("../Pagination");
const Text_1 = require("../Text");
function sortData(data, sortBy, sortDirection) {
    return [...data].sort((a, b) => {
        let left = (0, react_innertext_1.default)(a[sortBy]);
        let right = (0, react_innertext_1.default)(b[sortBy]);
        if (sortBy === 'date') {
            left = (0, helpers_1.getElementProperty)(a[sortBy], { type: 'time', property: 'dateTime' }) || left;
            right = (0, helpers_1.getElementProperty)(b[sortBy], { type: 'time', property: 'dateTime' }) || right;
        }
        if (sortDirection === 'desc') {
            return right.toLowerCase().localeCompare(left.toLowerCase());
        }
        return left.toLowerCase().localeCompare(right.toLowerCase());
    });
}
function DataTable(props) {
    const { breakpoint = 768, clean, columns, data, defaultColumn, disableScroll, loading = false, maxRows = 10, noResults, onClickPage, onClickSort, pagination = true, paginationCurrentPage, paginationServer, paginationTotalPages, responsive = false, scrollDuration = 400, scrollElement, scrollMargin = 16, width, ...rest } = props;
    const { darkMode = false } = (0, react_2.useTheme)();
    const element = (0, react_1.useRef)(null);
    const sortByDefault = defaultColumn || columns?.[0].key;
    const [{ currentPage, sortBy, sortDirection }, setState] = (0, react_use_1.useSetState)({
        currentPage: 1,
        sortBy: sortByDefault,
        sortDirection: 'asc',
    });
    (0, react_use_1.useUpdateEffect)(() => {
        const minLength = currentPage * maxRows - maxRows;
        if (data.length < minLength) {
            setState({ currentPage: 1 });
        }
    }, [currentPage, data.length, maxRows, setState]);
    const isResponsive = responsive && (width ?? window.innerWidth) < breakpoint;
    const totalPages = Math.ceil(data.length / maxRows);
    const handleClickPage = (event) => {
        const { page } = event.currentTarget.dataset;
        const pageNumber = Number(page);
        const scrollTarget = scrollElement || element.current;
        if (onClickPage) {
            onClickPage(pageNumber, paginationTotalPages || totalPages);
        }
        if (scrollTarget && !disableScroll) {
            (0, animations_1.scrollTo)(scrollTarget.getBoundingClientRect().top - scrollMargin, { scrollDuration });
        }
        if (paginationServer) {
            return;
        }
        setState({ currentPage: pageNumber });
    };
    const handleClickSort = (event) => {
        const { direction, name = '' } = event.currentTarget.dataset;
        const reverseDirection = direction === 'asc' ? 'desc' : 'asc';
        const nextDirection = sortBy === name ? reverseDirection : 'asc';
        const options = {
            sortBy: name,
            sortDirection: nextDirection,
        };
        if (onClickSort) {
            onClickSort(name, nextDirection);
        }
        if (paginationServer) {
            setState(options);
            return;
        }
        setState(options);
    };
    const isEmpty = !loading && !data.length;
    const rows = (0, react_1.useMemo)(() => {
        return paginationServer ? data : sortData(data, sortBy, sortDirection);
    }, [data, paginationServer, sortBy, sortDirection]);
    const body = (0, react_1.useMemo)(() => {
        if (isEmpty) {
            return ((0, jsx_runtime_1.jsx)(Box_1.BoxCenter, { padding: "md", radius: "sm", variant: "white", width: "100%", children: noResults || (0, jsx_runtime_1.jsx)(Text_1.Text, { bold: true, children: "Nothing found" }) }));
        }
        return ((0, jsx_runtime_1.jsx)(Body_1.default, { clean: clean, columns: columns, data: rows.slice(maxRows * (currentPage - 1), maxRows * currentPage), defaultColumn: defaultColumn, isResponsive: isResponsive, loading: loading }));
    }, [
        clean,
        columns,
        currentPage,
        defaultColumn,
        isEmpty,
        isResponsive,
        loading,
        maxRows,
        noResults,
        rows,
    ]);
    const styles = {};
    if (!clean) {
        styles.padding = 'md';
        styles.shade = darkMode ? 'darker' : 'lightest';
        styles.variant = 'gray';
    }
    return ((0, jsx_runtime_1.jsxs)(Box_1.Box, { ref: element, "data-component-name": "DataTable", maxWidth: "100%", radius: "xxs", width: width, ...styles, ...rest, children: [(0, jsx_runtime_1.jsx)(Head_1.default, { clean: clean, columns: columns, isDisabled: loading || isEmpty, isResponsive: isResponsive, onClick: handleClickSort, sortBy: sortBy, sortDirection: sortDirection }), body, pagination && ((0, jsx_runtime_1.jsx)(Box_1.Box, { border: clean ? [{ side: 'top' }] : undefined, pt: clean ? 'sm' : undefined, children: (0, jsx_runtime_1.jsx)(Pagination_1.Pagination, { currentPage: paginationCurrentPage || currentPage, onClick: handleClickPage, totalPages: paginationTotalPages || totalPages }) }))] }));
}
exports.DataTable = DataTable;
DataTable.defaultProps = {
    breakpoint: 768,
    clean: false,
    loading: false,
    maxRows: 10,
    pagination: true,
    responsive: false,
    scrollDuration: 400,
    scrollMargin: 16,
};
