// @gilbarbara/components/lib/Tabs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tab = exports.Tabs = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const react_use_1 = require("react-use");
const react_2 = require("@emotion/react");
const styled_1 = require("@emotion/styled");
const helpers_1 = require("@gilbarbara/helpers");
const Tab_1 = require("./Tab");
const ButtonBase_1 = require("../ButtonBase");
const Loader_1 = require("../Loader");
const helpers_2 = require("../modules/helpers");
const system_1 = require("../modules/system");
const NonIdealState_1 = require("../NonIdealState");
const StyledTabs = (0, styled_1.default)('div', (0, system_1.getStyledOptions)())(props => {
    const { direction } = props;
    return (0, react_2.css) `
    display: ${direction === 'vertical' ? 'block' : 'flex'};
    ${(0, system_1.marginStyles)(props)};
  `;
});
const StyledMenu = (0, styled_1.default)('div', (0, system_1.getStyledOptions)())(props => {
    const { direction, width } = props;
    const { grayLighter, spacing } = (0, helpers_2.getTheme)(props);
    const isHorizontal = direction === 'horizontal';
    const isVertical = direction === 'vertical';
    return (0, react_2.css) `
    align-items: flex-start;
    display: flex;
    flex-direction: ${isVertical ? 'row' : 'column'};
    margin-bottom: ${isVertical ? spacing.md : undefined};
    margin-right: ${isHorizontal ? spacing.md : undefined};
    max-width: ${width ? (0, helpers_2.px)(width) : undefined};
    overflow: ${isVertical ? 'auto hidden' : undefined};
    position: relative;

    &:before {
      border-bottom: ${isVertical ? `1px solid ${grayLighter}` : undefined};
      border-right: ${isHorizontal ? `1px solid ${grayLighter}` : undefined};
      bottom: 0;
      content: '';
      left: ${isVertical ? 0 : undefined};
      position: absolute;
      right: ${isHorizontal ? '-1px' : 0};
      top: ${isHorizontal ? 0 : undefined};
    }
  `;
});
const StyledMenuItem = (0, styled_1.default)(ButtonBase_1.ButtonBase, (0, system_1.getStyledOptions)('disabled'))(props => {
    const { direction, disableActiveBorderRadius, disabled, isActive, shade, variant = 'primary', } = props;
    const { grayDarker, grayMid, grayScale, spacing, variants } = (0, helpers_2.getTheme)(props);
    const darkMode = (0, system_1.isDarkMode)(props);
    const { bg } = (0, helpers_2.getColorVariant)(variant, shade, variants);
    let color = darkMode ? grayScale['20'] : grayDarker;
    const isVertical = direction === 'vertical';
    const isHorizontal = direction === 'horizontal';
    if (disabled) {
        color = grayMid;
    }
    else if (isActive) {
        color = bg;
    }
    return (0, react_2.css) `
    color: ${disabled ? grayMid : color};
    cursor: ${disabled ? 'not-allowed' : 'pointer'};
    line-height: 1;
    padding: ${spacing.xs} ${spacing.md};
    position: relative;
    width: ${isHorizontal ? '100%' : undefined};
    white-space: nowrap;

    ${isActive &&
        (0, react_2.css) `
      &:before {
        background-color: ${bg};
        bottom: ${isVertical ? '-1px' : 0};
        content: '';
        display: block;
        height: ${isVertical ? '3px' : undefined};
        left: ${isVertical ? 0 : undefined};
        position: absolute;
        right: ${isHorizontal ? '-1px' : 0};
        top: ${isHorizontal ? 0 : undefined};
        width: ${isHorizontal ? '3px' : undefined};

        ${!disableActiveBorderRadius &&
            (0, react_2.css) `
          border-top-left-radius: 3px;
          border-top-right-radius: ${isVertical ? '3px' : undefined};
          border-bottom-left-radius: ${isHorizontal ? '3px' : undefined};
        `};
      }
    `}
  `;
});
const StyledContent = (0, styled_1.default)('div', (0, system_1.getStyledOptions)())(props => {
    const { maxHeight, minHeight } = props;
    return (0, react_2.css) `
    max-height: ${maxHeight ? (0, helpers_2.px)(maxHeight) : undefined};
    min-height: ${minHeight ? (0, helpers_2.px)(minHeight) : undefined};
    overflow-y: auto;
  `;
});
function Tabs(props) {
    const { children, initialId = '', loader, maxHeight, minHeight, noContent, onClick, shade, variant, ...rest } = props;
    const [{ activeId, error, isReady, tabs, width }, setState] = (0, react_use_1.useSetState)({
        activeId: initialId,
        error: false,
        isReady: false,
        tabs: [],
        width: null,
    });
    const isMounted = (0, react_1.useRef)(false);
    const uniqueId = (0, react_1.useRef)((0, helpers_1.unique)(6));
    const [ref, measurements] = (0, react_use_1.useMeasure)();
    (0, react_1.useEffect)(() => {
        isMounted.current = true;
        return () => {
            isMounted.current = false;
        };
    }, []);
    (0, react_1.useEffect)(() => {
        const nextState = {
            error: false,
            isReady: true,
            tabs: [],
        };
        react_1.Children.forEach(children, (child, index) => {
            if (!(0, react_1.isValidElement)(child)) {
                return;
            }
            if (child.type === Tab_1.Tab) {
                nextState.tabs.push((0, helpers_1.omit)(child.props, 'children'));
                if (!activeId && index === 0) {
                    nextState.activeId = child.props.id;
                }
            }
            else {
                nextState.error = true;
                // eslint-disable-next-line no-console
                console.warn('Invalid children detected. Only the Tab component is allowed.');
            }
        });
        setState(nextState);
    }, [activeId, children, setState]);
    (0, react_1.useEffect)(() => {
        if (measurements.width && measurements.width !== width) {
            setState({ width: measurements.width });
        }
    }, [measurements, setState, width]);
    const handleClickItem = (event) => {
        const { disabled, id = '' } = event.currentTarget.dataset;
        if (activeId === id || disabled === 'true') {
            return;
        }
        setState({ activeId: id });
        if (onClick) {
            onClick(id);
        }
    };
    if (error) {
        return null;
    }
    const content = {};
    if (isReady) {
        if (tabs.length) {
            if (width || rest.direction === 'horizontal') {
                content.menu = ((0, jsx_runtime_1.jsx)(StyledMenu, { "data-component-name": "TabsMenu", direction: rest.direction, role: "tablist", width: width, children: tabs.map(d => ((0, jsx_runtime_1.jsx)(StyledMenuItem, { "aria-controls": `panel-${uniqueId.current}-${d.id}`, "aria-selected": d.id === activeId, "data-component-name": "TabsMenuItem", "data-disabled": !!d.disabled, "data-id": d.id, direction: rest.direction, disableActiveBorderRadius: rest.disableActiveBorderRadius, disabled: !!d.disabled, isActive: d.id === activeId, onClick: handleClickItem, role: "tab", shade: shade, variant: variant, children: d.title }, d.id))) }));
            }
            content.main = ((0, jsx_runtime_1.jsx)(StyledContent, { "data-component-name": "TabsContent", maxHeight: maxHeight, minHeight: minHeight, children: react_1.Children.toArray(children)
                    .filter(d => (0, react_1.isValidElement)(d) && d.props.id === activeId)
                    .map(d => (0, react_1.isValidElement)(d) && ((0, jsx_runtime_1.jsx)("div", { "data-component-name": "TabsPanel", id: `panel-${uniqueId.current}-${d.props.id}`, role: "tabpanel", children: d.props.children }, d.props.id))) }));
        }
        else {
            content.main = noContent || ((0, jsx_runtime_1.jsx)(NonIdealState_1.NonIdealState, { description: "Content not available", icon: "danger", size: "sm" }));
        }
    }
    else {
        content.main = loader || (0, jsx_runtime_1.jsx)(Loader_1.Loader, {});
    }
    return ((0, jsx_runtime_1.jsxs)(StyledTabs, { "data-component-name": "Tabs", ...rest, children: [rest.direction === 'vertical' && (0, jsx_runtime_1.jsx)("div", { ref: ref }), content.menu, content.main] }));
}
exports.Tabs = Tabs;
Tabs.defaultProps = {
    direction: 'vertical',
    disableActiveBorderRadius: false,
    shade: 'mid',
    variant: 'primary',
};
var Tab_2 = require("./Tab");
Object.defineProperty(exports, "Tab", { enumerable: true, get: function () { return Tab_2.Tab; } });
