// @gilbarbara/components/lib/modules/colors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getGrayScale = exports.getColorScale = void 0;
const helpers_1 = require("@gilbarbara/helpers");
const colorizr_1 = require("colorizr");
function getColorScale(color) {
    const { h, l, s } = (0, colorizr_1.hex2hsl)(color);
    const lightest = (0, colorizr_1.hsl2hex)({ h, s, l: (0, helpers_1.rangeLimit)(l * 1.9, 92, 96) });
    const lighter = (0, colorizr_1.hsl2hex)({ h, s, l: (0, helpers_1.rangeLimit)(l * 1.6, 82, 92) });
    const light = (0, colorizr_1.hsl2hex)({ h, s, l: (0, helpers_1.rangeLimit)(l * 1.4, 64, 72) });
    const dark = (0, colorizr_1.hsl2hex)({ h, s, l: (0, helpers_1.rangeLimit)(l * 0.7, 32) });
    const darker = (0, colorizr_1.hsl2hex)({ h, s, l: (0, helpers_1.rangeLimit)(l * 0.45, 16) });
    const darkest = (0, colorizr_1.hsl2hex)({ h, s, l: (0, helpers_1.rangeLimit)(l * 0.3, 8) });
    return {
        lightest: { bg: lightest, color: (0, colorizr_1.textColor)(lightest) },
        lighter: { bg: lighter, color: (0, colorizr_1.textColor)(lighter) },
        light: { bg: light, color: (0, colorizr_1.textColor)(light) },
        mid: { bg: color, color: (0, colorizr_1.textColor)(color) },
        dark: { bg: dark, color: (0, colorizr_1.textColor)(dark) },
        darker: { bg: darker, color: (0, colorizr_1.textColor)(darker) },
        darkest: { bg: darkest, color: (0, colorizr_1.textColor)(darkest) },
    };
}
exports.getColorScale = getColorScale;
function getGrayScale(lightest, lighter, light, mid, dark, darker, darkest) {
    return {
        lightest: { bg: lightest, color: (0, colorizr_1.textColor)(lightest) },
        lighter: { bg: lighter, color: (0, colorizr_1.textColor)(lighter) },
        light: { bg: light, color: (0, colorizr_1.textColor)(light) },
        mid: { bg: mid, color: (0, colorizr_1.textColor)(mid) },
        dark: { bg: dark, color: (0, colorizr_1.textColor)(dark) },
        darker: { bg: darker, color: (0, colorizr_1.textColor)(darker) },
        darkest: { bg: darkest, color: (0, colorizr_1.textColor)(darkest) },
    };
}
exports.getGrayScale = getGrayScale;
