// @gilbarbara/components/lib/modules/helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.responsive = exports.recursiveChildrenEnhancer = exports.px = exports.mergeTheme = exports.isCSSUnit = exports.getTheme = exports.getMediaQueries = exports.getElementProperty = exports.getColorVariant = exports.createMediaQuery = exports.clearNumber = void 0;
const react_1 = require("react");
const react_2 = require("@emotion/react");
const helpers_1 = require("@gilbarbara/helpers");
const deepmerge_ts_1 = require("deepmerge-ts");
const is_lite_1 = require("is-lite");
const colors_1 = require("./colors");
const theme = require("./theme");
const { black, breakpoints, white } = theme;
const deepmerge = (0, deepmerge_ts_1.deepmergeCustom)({
    mergeArrays: false,
});
function clearNumber(value) {
    return value.replace(/\D+/g, '');
}
exports.clearNumber = clearNumber;
/**
 * Generate the media query
 */
function createMediaQuery(size, mediaQueries) {
    if (isCSSUnit(size) || is_lite_1.default.numericString(size)) {
        return `@media screen and (min-width: ${px(size)})`;
    }
    return mediaQueries[size];
}
exports.createMediaQuery = createMediaQuery;
/**
 * Get color from theme
 */
function getColorVariant(variant, shade = 'mid', variants = theme.variants) {
    try {
        switch (variant) {
            case 'black': {
                return { bg: black, color: white };
            }
            case 'white': {
                return { bg: white, color: black };
            }
            default: {
                return variants[variant][shade];
            }
        }
    }
    catch {
        return variants.primary.mid;
    }
}
exports.getColorVariant = getColorVariant;
function getElementProperty(element, options) {
    const { children } = element.props;
    const { property, type } = options;
    const getValue = (input) => {
        const { props } = input;
        if (input.type === type) {
            if (property) {
                return props?.[property] || null;
            }
            return props.children || null;
        }
        return null;
    };
    for (const child of react_1.Children.toArray(children)) {
        if (!(0, react_1.isValidElement)(child)) {
            return getValue(element);
        }
        const childrenElement = child.props?.children?.type === type ? child.props.children : undefined;
        const selectedElement = child.type === type ? child : childrenElement;
        if (selectedElement) {
            return getValue(selectedElement);
        }
        if (is_lite_1.default.array(child.props.children)) {
            return getElementProperty(child, options);
        }
    }
    return null;
}
exports.getElementProperty = getElementProperty;
function getMediaQueries() {
    return Object.keys(breakpoints)
        .filter(d => Number.isNaN(parseInt(d, 10)))
        .reduce((acc, d) => {
        acc[d] = `@media screen and (min-width: ${px(breakpoints[d])})`;
        return acc;
    }, {});
}
exports.getMediaQueries = getMediaQueries;
function getTheme(props) {
    return mergeTheme(props?.theme || {});
}
exports.getTheme = getTheme;
function isCSSUnit(value) {
    const units = ['em', 'px', 'rem', 'vh', 'vmax', 'vmin', 'vw'];
    const regex = new RegExp(`\\d+(${units.join('|')})$`);
    return typeof value === 'string' && regex.test(value);
}
exports.isCSSUnit = isCSSUnit;
function mergeTheme(customTheme = {}) {
    const nextTheme = deepmerge({ ...theme }, customTheme);
    const { gray, grayDark, grayDarker, grayDarkest, grayLight, grayLighter, grayLightest } = nextTheme;
    const baseVariants = Object.entries({ ...nextTheme.colors, gray: nextTheme.gray })
        .filter(([key]) => !['black', 'white'].includes(key))
        .reduce((acc, [key, value]) => {
        acc[key] =
            key === 'gray'
                ? (0, colors_1.getGrayScale)(grayLightest, grayLighter, grayLight, gray, grayDark, grayDarker, grayDarkest)
                : (0, colors_1.getColorScale)(value);
        return acc;
    }, {});
    return {
        ...nextTheme,
        variants: deepmerge(baseVariants, customTheme.variants || {}),
    };
}
exports.mergeTheme = mergeTheme;
function px(value) {
    return is_lite_1.default.number(value) || is_lite_1.default.numericString(value) ? `${value}px` : value;
}
exports.px = px;
function recursiveChildrenEnhancer(children, props, options) {
    const { componentType, overrideProps } = options;
    return react_1.Children.map(children, child => {
        if (!(0, react_1.isValidElement)(child)) {
            return child;
        }
        const nextProps = overrideProps ? props : (0, helpers_1.omit)(props, ...Object.keys(child.props));
        if (child.props.children) {
            let childProps = {
                children: is_lite_1.default.function(child.props.children)
                    ? child.props.children
                    : recursiveChildrenEnhancer(child.props.children, nextProps, options),
            };
            if (child.type === componentType) {
                childProps = { ...childProps, ...nextProps };
            }
            return (0, react_1.cloneElement)(child, childProps);
        }
        if (child.type !== componentType) {
            return child;
        }
        return (0, react_1.cloneElement)(child, nextProps);
    });
}
exports.recursiveChildrenEnhancer = recursiveChildrenEnhancer;
/**
 * Helper to generate responsive media queries
 */
function responsive(rules) {
    const entries = {};
    const mediaQueries = getMediaQueries();
    for (const rule in rules) {
        /* istanbul ignore else */
        if ({}.hasOwnProperty.call(rules, rule)) {
            const breakpoint = rule;
            const styles = rules[rule];
            const query = createMediaQuery(breakpoint, mediaQueries);
            if (breakpoint === '_') {
                Object.entries(styles).forEach(([k, v]) => {
                    entries[k] = v;
                });
            }
            else if (query) {
                entries[query] = styles;
            }
        }
    }
    return (0, react_2.css)(entries);
}
exports.responsive = responsive;
